docs_headless/src/content/docs/guides/Architecture.md

```diff
@@ -0,0 +1,304 @@
+---
+title: "Key Concepts"
+---
+
+React-admin relies on a several design decisions that structure its codebase.
+
+## Single-Page Application
+
+React-admin is specifically designed to build [Single-Page Applications (SPA)](https://en.wikipedia.org/wiki/Single-page_application). In a react-admin app, the browser fetches the required HTML, CSS, and JavaScript to render the application only once. Subsequently, data is fetched from APIs through AJAX calls. This is in contrast to traditional web applications, where the browser fetches a new HTML page for each screen.
+
+![SPA lifecycle](../../../assets/SPA-lifecycle.png)
+
+The SPA architecture ensures that react-admin apps are [exceptionally fast](./Features.md#fast), easy to host, and compatible with existing APIs without requiring a dedicated backend. 
+
+To achieve this, react-admin utilizes an internal router, powered by `react-router`, to display the appropriate screen when the user clicks on a link. Developers can define routes using the [`<Resource>`](./Resource.md) component for CRUD routes and the [`<CustomRoutes>`](./CustomRoutes.md) component for other routes.
+
+For example, the following react-admin application:
+
+```jsx
```

Comment:
I tried using syntax like `jsx{5}` but unfortunately it does not work. Do you think we could add support for it?